src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplay/ContentItemDisplayCoordinator.cs

```diff
@@ -77,22 +84,46 @@ public async Task BuildDisplayAsync(ContentItem contentItem, BuildDisplayContext
 
                 if (part != null)
                 {
-                    foreach (var displayDriver in _partDisplayDrivers)
+                    var partDisplayDrivers = _contentPartDisplayDriverResolver.GetDisplayDrivers(partTypeName);
+                    // For backward compatability check for Any().
+                    // TODO: Any() can be removed in a future release as the recommended way is to use ContentOptions.
+                    if (partDisplayDrivers != null && partDisplayDrivers.Any())
                     {
-                        try
+                        foreach (var partDisplayDriver in partDisplayDrivers)
                         {
-                            var result = await displayDriver.BuildDisplayAsync(part, contentTypePartDefinition, context);
-                            if (result != null)
+                            try
                             {
-                                await result.ApplyAsync(context);
+                                var result = await partDisplayDriver.BuildDisplayAsync(part, contentTypePartDefinition, context);
+                                if (result != null)
+                                {
+                                    await result.ApplyAsync(context);
+                                }
+                            }
+                            catch (Exception ex)
+                            {
+                                InvokeExtensions.HandleException(ex, Logger, partDisplayDrivers.GetType().Name, nameof(BuildDisplayAsync));
                             }
                         }
-                        catch (Exception ex)
+                    }
+                    // TODO: This can be removed in a future release as the recommended way is to use ContentOptions.
+                    else
+                    {
+                        foreach (var displayDriver in _partDisplayDrivers)
                         {
-                            InvokeExtensions.HandleException(ex, Logger, displayDriver.GetType().Name, "BuildDisplayAsync");
+                            try
```

Comment:
@sebastienros on your recent pr https://github.com/OrchardCMS/OrchardCore/pull/4887 you've reduced the allocation pattern for using the invoke extensions.
Just noting that the current `BuildDisplayAsync` method doesn't use the 
`await partDisplayDrivers.InvokeAsync(async (driver, part, typePartDefinition, context) =>` technique at all, it is only the `BuildEditorAsync` that used it previously.
So if there is an opportunity to reduce allocations on `
BuildDisplayAsync` (regardless of whether this PR goes through), perhaps we should use them?